### PR TITLE
feat: add messages api

### PR DESCRIPTION
**Pull request by namchuai**
## Describe Your Changes

- Adding messages API which compatible to https://platform.openai.com/docs/api-reference/messages

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/common/api-dto/messages/delete_message_response.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "common/json_serializable.h"
+
+namespace ApiResponseDto {
```
Comment (vansangpfiev) [inline engine/common/api-dto/messages/delete_message_response.h]: I think we use snake_case for namespace

File: engine/common/message.h [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  // The Unix timestamp (in seconds) for when the message was completed.
+  std::optional<uint32_t> completed_at;
+
+  // The Unix timestamp (in seconds) for when the message was marked as incomplete.
+  std::optional<uint32_t> incomplete_at;
+
+  Role role;
+
+  // The content of the message in array of text and/or images.
+  std::vector<std::unique_ptr<Content>> content;
```
Comment (vansangpfiev) [inline engine/common/message.h:61]: Not sure why do we need `unique_ptr`?

Comment (namchuai) [reply engine/common/message.h:61]: because `Content` is a pure virtual class. c++ STL containers are not able to handle polymorphic

**Comment by vansangpfiev** [comment]
Great work!
It would be nice if we have unit tests for `ToJson` functions.
Also, I think can consider moving the implementation to the cc files to reduce the compilation time, especially on Windows.

**Review by vansangpfiev — APPROVED** [review]

**Comment by namchuai** [comment]
I will add the test in the following PR

**Pull request merged**